src/components/Header/Header.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { Ref } from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+import { updateTodo } from '../../api/todos';
+import { Errors } from '../../types/Errors';
+
+type Props = {
+  onSubmit: (e: React.FormEvent) => void;
+  onChange: (value: string) => void;
+  inputRef: Ref<HTMLInputElement> | null;
+  onUpdateTodos: (newTodo: Todo) => void;
+  newTitle: string;
+  isLoading: boolean;
+  todos: Todo[];
+  onError: (error: Errors) => void;
+  setLoadingTodoIds: (ids: number[]) => void;
+};
+
+export const Header: React.FC<Props> = ({
+  onSubmit,
+  onChange,
+  inputRef,
+  onUpdateTodos,
+  newTitle,
+  isLoading,
+  todos,
+  onError,
+  setLoadingTodoIds,
+}) => {
+  const isEveryTodoctive =
+    todos.every(todo => todo.completed) && !!todos.length;
+
+  const handleToggleAll = async () => {
+    const filteredTasks = todos.filter(todo => !todo.completed);
+    const unfinishedTasks = filteredTasks.length ? filteredTasks : todos;
+
+    try {
+      setLoadingTodoIds([...unfinishedTasks.map(task => task.id)]);
```

Comment:
Consider moving this to a variable `unfinishedTasks.map(task => task.id)` and then spread